### PR TITLE
feat: do not use finalizer to stop metrics cache loop

### PR DESCRIPTION
**Pull request by whynowy**
Signed-off-by: Derek Wang <whynowy@gmail.com>

### CODE REVIEW

File: manager/controllers/scaling/metrics_cache.go [hunk 1]
```diff
@@ -98,6 +109,43 @@ func (m *MetricsCacheHandler) Start(ctx context.Context) {
 		go m.pullMetrics(ctx, i, keyCh)
 	}
 
+	go func(cctx context.Context) {
```
Comment (alexec) [inline manager/controllers/scaling/metrics_cache.go:112]: I'd use JitterWithContext

**Comment by alexec** [comment]
How does that interplay with peeking?

**Comment by whynowy** [comment]
> How does that interplay with peeking?

I don't want to clean up the key from the dead keys queue when successfully hitting the endpoint during peeking - it's unnecessary to do so which will cost a little bit to check the map, just leave it to the GoRoutine to check Step existence.

**Comment by alexec** [comment]
Sure. When we get an informer re-sync (every 20m) am I right it thinking it is likely to put any key incorrectly removed back into the map?

File: manager/controllers/scaling/metrics_cache.go (hunk 1 above)
Comment (whynowy) [reply manager/controllers/scaling/metrics_cache.go:112]: It does not need to have a jitter.

Comment (alexec) [reply manager/controllers/scaling/metrics_cache.go:112]: Panic recovery.

**Pull request merged**